File: core/src/main/java/com/drevelopment/couponcodes/core/entity/SimplePlayer.java
package com.drevelopment.couponcodes.core.entity;

import com.drevelopment.couponcodes.api.CouponCodes;
import com.drevelopment.couponcodes.api.command.CommandSender;
import com.drevelopment.couponcodes.api.entity.Player;

public abstract class SimplePlayer implements CommandSender, Player {

	/**
	 * Gets whether the player has a certain permission node
	 */
	public boolean hasPermission(String node) {
		return CouponCodes.getPermissionHandler().hasPermission(this, node);
	}

	public String getName() {
		return null;
	}

	public String getUUID() {
		return null;
	}

	public abstract void setLevel(int level);

	public abstract int getLevel();

	public abstract void giveItem(int item, int amount);

	public abstract String getLocale();

}

File: core/src/main/kotlin/tech/feldman/couponcodes/core/entity/SimplePlayer.kt
/**
 * The MIT License
 * Copyright (c) 2015 Nicholas Feldman (AngrySoundTech)
 * <p/>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p/>
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * <p/>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package tech.feldman.couponcodes.core.entity

import tech.feldman.couponcodes.api.CouponCodes
import tech.feldman.couponcodes.api.command.CommandSender
import tech.feldman.couponcodes.api.entity.Player
import tech.feldman.couponcodes.api.exceptions.UnknownMaterialException
import java.util.*

abstract class SimplePlayer : CommandSender, Player {

    override fun hasPermission(node: String) = CouponCodes.getPermissionHandler().hasPermission(this, node)

    override fun getName(): String? = null

    override fun getUUID(): String? = null

    abstract override fun getLocale(): Locale

    abstract override fun getLevel(): Int

    abstract override fun setLevel(level: Int)

    @Throws(UnknownMaterialException::class)
    abstract override fun giveItem(item: String, amount: Int)

}
